src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/Transition.kt

```diff
@@ -0,0 +1,148 @@
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.ScheduleParser
+import org.elasticsearch.ElasticsearchParseException
+import org.elasticsearch.common.unit.ByteSizeValue
+import org.elasticsearch.common.unit.TimeValue
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.ToXContentObject
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import java.io.IOException
+
+data class Transition(
+    val stateName: String,
+    val conditions: Conditions?
+) : ToXContentObject {
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        builder.startObject()
+            .field(STATE_FIELD, stateName)
+        if (conditions != null) builder.field(CONDITIONS_FIELD, conditions)
+        return builder.endObject()
+    }
+
+    companion object {
+        const val STATE_FIELD = "state_name"
+        const val CONDITIONS_FIELD = "conditions"
+
+        @JvmStatic
+        @Throws(IOException::class)
+        fun parse(xcp: XContentParser): Transition {
+            lateinit var name: String
+            var conditions: Conditions? = null
+
+            ensureExpectedToken(Token.START_OBJECT, xcp.currentToken(), xcp::getTokenLocation)
+            while (xcp.nextToken() != Token.END_OBJECT) {
+                val fieldName = xcp.currentName()
+                xcp.nextToken()
+
+                when (fieldName) {
+                    STATE_FIELD -> name = xcp.text()
+                    CONDITIONS_FIELD -> conditions = Conditions.parse(xcp)
+                    else -> throw IllegalArgumentException("Invalid field: [$fieldName] found in Transition.")
+                }
+            }
+
+            return Transition(
+                stateName = requireNotNull(name) { "Transition state name is null" },
+                conditions = conditions
+            )
+        }
+    }
+}
+
+data class Conditions(
+    val indexAge: String? = null,
+    val docCount: Long? = null,
+    val size: String? = null,
+    val cron: CronSchedule? = null
+) : ToXContentObject {
+
+    init {
+        val conditionsList = listOf(indexAge, docCount, size, cron)
+        require(conditionsList.filterNotNull().size == 1) { "Cannot provide more than one Transition condition" }
+
+        // Validate indexAge condition
+        if (indexAge != null) {
+            try {
+                TimeValue.parseTimeValue(indexAge, "")
+            } catch (e: IllegalArgumentException) {
+                throw IllegalArgumentException("Must have a valid index age Transition condition")
+            }
+        }
+
+        // Validate size condition
+        if (size != null) {
+            try {
+                ByteSizeValue.parseBytesSizeValue(size, "")
```

Comment:
It seems like ByteSizeValue.parseBytesSize accepts "0" and "-1" for size, do we want to allow these values?

Reply:
Yeah I noticed that these were considered special values to ```ByteSizeValue```. I can't really see a use for them. If you agree, I can include additional validation to make sure size is > 0.

Reply:
Agreed, sounds good

Reply:
Done.